crates/matrix-sdk-crypto/src/olm/group_sessions/inbound.rs

```diff
@@ -55,6 +55,26 @@ use crate::{
     },
 };
 
+/// The trust level required to decrypt an event
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+pub enum TrustRequirement {
+    /// Decrypt events from everyone regardless of trust
+    Untrusted,
+    /// Only decrypt events from cross-signed or legacy devices
+    CrossSignedOrLegacy,
+    /// Only decrypt events from cross-signed devices
+    CrossSigned,
+    /// Only decrypt events from cross-signed devices where we have verified the user
+    VerifiedUser,
+}
+
+/// Settings for decrypting messages
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct DecryptionSettings {
```

Comment:
I initially put this in this file for symmetry with `EncryptionSettings`, which is in `olm/group_sessions/outbound.rs`, but it only gets used in `machine.rs`.  So I suspect that I should put this in `machine.rs` instead?  Or is there a better place to put it?

Reply:
Probably into the `types/` submodule.

Reply:
That sounds reasonable.  I don't really see an existing file in there that looks like a compelling place to put it.  Should I just put it in `types/mod.rs`, or create a new file, like `types/decryption.rs`?

Reply:
A new file sounds fine.